### PR TITLE
feat: Allow skipping `cdk bootstrap`

### PR DESCRIPTION
**Pull request by rickselby**
Most of our cdk pipelines don't need the bootstrapped resources, so it would be useful to optionally skip the bootstrap step.

Caveat: I don't write a lot of javascript, certainly nothing like this, so if I've gone about this the wrong way, I apologise - let me know what needs changing and I'll make the required tweaks.

### CODE REVIEW

**Review by IPWright83 — APPROVED** [review]
Looks fine by me :)

**Comment by ioannisGiak89** [comment]
Hi @rickselby, when do you actually need to run bootstrap and when not?

File: src/cdkCommands.test.js [hunk 1]
```diff
@@ -13,9 +13,25 @@ describe('cdkCommands', () => {
...
+        await runCdkCommands({
+            AWS_ACCOUNT_ID: 'account_id',
+            AWS_REGION: 'eu-west-1',
+            AWS_ROLE_NAME: 'role_name',
+            AWS_STACK_NAME: 'stack_name',
+            INFRASTRUCTURE_PATH: 'some/path/',
+            SKIP_BOOTSTRAP: 'true',
+        });
+
+        expect(execa.mock.calls[2]).toMatchSnapshot();
```
Comment (natseg) [inline src/cdkCommands.test.js:35]: So for this additional test, we need to take into account that `jest` keeps track of the mock calls unless we explicitly clear then in a [hook](https://jestjs.io/docs/en/setup-teardown#repeating-setup-for-many-tests).
```suggestion
   beforeEach(() => jest.clearAllMocks()) // on line 7
```

```suggestion
    expect(execa).toHaveBeenCalledTimes(1);
	expect(execa.mock.calls[0]).toMatchSnapshot();
```

**Review by natseg — CHANGES_REQUESTED** [review]
Very good work, thank you so much, just a small thing to fix in the test! @rickselby 
I appreciate that you've made this a non-breaking change :star:

File: src/cdkCommands.test.js (hunk 1 above)
Comment (rickselby) [reply src/cdkCommands.test.js]: Thanks, that's much better!

**Comment by rickselby** [comment]
> Hi @rickselby, when do you actually need to run bootstrap and when not?

I don't think we ever actually run bootstrap - we're using cdk to deploy infrastructure, rather than apps.

File: src/cdkCommands.test.js (hunk 1 above)
Comment (IPWright83) [reply src/cdkCommands.test.js]: Doh, sorry I missed that @natseg

**Review by natseg — APPROVED** [review]
Nice one, thanks!

**Review by natseg — CHANGES_REQUESTED** [review]
I don't want to block this but as a failsafe @ben-eb did spot (good spot) the  changes to package-lock... I wonder what happened there? What happens when discarding changes to it and running `npm ci`? Sorry for missing it.

**Comment by rickselby** [comment]
That was a mistake, I think I have a newer version of node / npm that wanted to make the changes. I kept it out of the first commit but it slipped through later.

**Review by natseg — APPROVED** [review]
I think this time, we are good! Thanks so much for your patience 👍

**Pull request merged**